shell/platform/android/io/flutter/embedding/engine/dart/DartMessenger.java

```diff
@@ -215,12 +273,43 @@ public void handleMessageFromDart(
             Trace.endSection();
           }
         };
-    @NonNull
     final DartMessengerTaskQueue nonnullTaskQueue =
         taskQueue == null ? platformTaskQueue : taskQueue;
     nonnullTaskQueue.dispatch(myRunnable);
   }
 
+  @Override
+  public void handleMessageFromDart(
+      @NonNull String channel, @Nullable ByteBuffer message, int replyId, long messageData) {
+    // Called from the ui thread.
+    Log.v(TAG, "Received message from Dart over channel '" + channel + "'");
+
+    HandlerInfo handlerInfo;
+    boolean messageDeferred;
+    synchronized (handlersLock) {
+      handlerInfo = messageHandlers.get(channel);
+      messageDeferred = (enableBufferingIncomingMessages && handlerInfo == null);
+      if (messageDeferred) {
+        // The channel is not defined when the Dart VM sends a message before the channels are
+        // registered.
+        //
+        // This is possible if the Dart VM starts before channel registration, and if the thread
+        // that registers the channels is busy or slow at registering the channel handlers.
+        //
+        // In such cases, the task dispatchers are queued, and processed when the channel is
+        // defined.
```

Comment:
Can't this also happen if the handler gets unregistered?

Reply:
that is right. In this case, `enableBufferingIncomingMessages` should be set to `false`. That makes sense to add.